src/SmartComponents/AddSystemModal/AddSystemModal.js

```diff
@@ -153,19 +153,27 @@ export class AddSystemModal extends Component {
     }
 
     systemContentSelect = (data) => {
-        const { entities, handleSystemSelection } = this.props;
+        const { entities, handleSystemSelection, selectedSystemContent } = this.props;
         let selectedSystems = [];
 
         if (data.id === 0) {
             selectedSystems = entities.rows.map(function(row) {
                 return { id: row.id, name: row.display_name, icon: <ServerIcon /> };
             });
         } else {
-            entities.rows.forEach(function(row) {
-                if (row.id === data.id) {
-                    selectedSystems.push({ id: row.id, name: row.display_name, icon: <ServerIcon /> });
-                }
-            });
+            if (!data.selected) {
+                selectedSystemContent.forEach(function(system) {
```

Comment:
Observation: I guess you can use shortened syntax for anonymous functions:
```
(system) => { ...
```
babel should handle that in the resulting JS.